internal/runtimehandlerhooks/high_performance_hooks.go

```diff
@@ -124,3 +183,71 @@ func setCPUSLoadBalancing(c *oci.Container, enable bool, schedDomainDir string)
 
 	return nil
 }
+
+func setIRQLoadBalancing(c *oci.Container, enable bool) error {
+	if c.Spec().Linux == nil ||
+		c.Spec().Linux.Resources == nil ||
+		c.Spec().Linux.Resources.CPU == nil ||
+		c.Spec().Linux.Resources.CPU.Cpus == "" {
+		return fmt.Errorf("failed to find the container %q CPUs", c.ID())
+	}
+
+	content, err := ioutil.ReadFile(irqSmpAffinityFile)
+	if err != nil {
+		return err
+	}
+	currentIRQSMPSetting := strings.Trim(string(content), "\n")
+
+	newIRQSMPSetting, err := UpdateIRQSmpAffinityMask(c.Spec().Linux.Resources.CPU.Cpus, currentIRQSMPSetting, enable)
+	if err != nil {
+		return err
+	}
+	newIRQSMPSetting += "\n"
+	ioutil.WriteFile(irqSmpAffinityFile, []byte(newIRQSMPSetting), 0644)
+	return nil
+}
+
+func setCPUQuota(ctx context.Context, parentDir string, c *oci.Container, enable bool) error {
+	original := c.Spec().Linux.CgroupsPath
+	var rpath string
+	if strings.Contains(original, ":") {
+		// systemd managed, first replace the colon in "...slice:" with "/"
+		step1 := strings.Replace(original, ":", "/", 1)
+		// then replace colon in "crio:" with "-"
+		step2 := strings.Replace(step1, ":", "-", 1)
+		// lastly append ".scope"
+		rpath = step2 + ".scope"
+	} else {
+		// cgroup manageed
+		rpath = original
+	}
+
+	// cgroup v1, v2, systemd, cgroupfs
+	var prefixes = []string{"/sys/fs/cgroup/cpu/",
+		"/sys/fs/cgroup/cpu/kubepods.slice/",
+		"/sys/fs/cgroup/cpu/kubepods/",
+		"/sys/fs/cgroup/",
+		"/sys/fs/cgroup/kubepods.slice/",
+		"/sys/fs/cgroup/kubepods/",
+	}
+	var cfsQuotaFile string
+	var parentCfsQuotaFile string
+	for _, prefix := range prefixes {
+		cfsQuotaFile = prefix + rpath + "/cpu.cfs_quota_us"
+		if _, err := os.Stat(cfsQuotaFile); err == nil {
+			parentCfsQuotaFile = prefix + parentDir + "/cpu.cfs_quota_us"
+			log.Infof(ctx, "Update %q for the container %q", cfsQuotaFile, c.ID())
+			if enable {
+				// there should not have use case to get here, as the pod cgroup will be deleted when the pod end
+				ioutil.WriteFile(cfsQuotaFile, []byte("0\n"), 0644)
+				ioutil.WriteFile(parentCfsQuotaFile, []byte("0\n"), 0644)
+			} else {
+				ioutil.WriteFile(cfsQuotaFile, []byte("-1\n"), 0644)
+				ioutil.WriteFile(parentCfsQuotaFile, []byte("-1\n"), 0644)
+			}
+			return nil
+		}
+	}
+	log.Infof(ctx, "Failed to find cpu.cfs_quota_us file for the container %q", c.ID())
+	return nil
+}
```

Comment:
I don't think writing directly to the cgroup is the right thing to do when using systemd.  systemd is the owner for the cgroup.  We should either use the dbus API to change these settings, or I'd suggest using either runc/ libcontainer/cgroups or containers/podman/pkg/cgroups